dimod/vartypes.py

```diff
@@ -55,13 +55,28 @@
 """
 import enum
 import typing
+import warnings
+
+from collections.abc import Container, Iterable
 
 __all__ = ['as_vartype',
            'Vartype', 'ExtendedVartype',
-           'SPIN', 'BINARY', 'DISCRETE',
+           'SPIN', 'BINARY', 'DISCRETE', 'INTEGER',
            ]
 
 
+class Integers(Container):
+    """Container for testing integer membership."""
+    def __contains__(self, item):
+        try:
+            return int(item) == item
+        except TypeError:
+            return False
+
+
+integers = Integers()
```

Comment:
I'm sure there's a good reason for this, but it's not obvious to me. Where is membership testing used?

Reply:
An example would be something like https://github.com/dwavesystems/dimod/blob/63d17243b2aab52852df8e4e8760150ee2bc40c2/dimod/core/bqm.py#L1372

Reply:
Of course. Enum's value is actually a domain of the type. Forgot that.